src/MongoDB.Driver/IMongoCollection.cs

```diff
@@ -862,12 +862,62 @@ public interface IMongoCollection<TDocument>
         /// </summary>
         /// <param name="filter">The filter.</param>
         /// <param name="replacement">The replacement.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>
+        /// The result of the replacement.
+        /// </returns>
+        ReplaceOneResult ReplaceOne(FilterDefinition<TDocument> filter, TDocument replacement, CancellationToken cancellationToken = default(CancellationToken));
```

Comment:
This overload is not necessary.

Instead we can use `options = null` on line 881.

Same for `ReplaceOneAsync`.

Reply:
Also, when I replaced parameters as you suggested: 
this code is not compatible anymore:
```
var opt1 = collection.ReplaceOne(filter, replacement, CancellationToken.None);
```
The error: 
```
CS1503	Argument 3: cannot convert from 'System.Threading.CancellationToken' to 'MongoDB.Driver.ReplaceOptions'	MongoDB.Driver.Examples (net452), MongoDB.Driver.Examples (netcoreapp2.1)
```

Reply:
I don't believe that code was compatible before either.

As a general rule optional parameters have advantages over multiple overloads, because the number of overloads grows exponentially when you don't use optional parameters.